### ISSUE TITLE
Parse error in Redshift dialect in CREATE TABLE expression with IDENTITY column

### ISSUE THREAD

**Issue by menzenski** [sqlfluff/sqlfluff#3429]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A `CREATE TABLE` expression with an `IDENTITY` column throws a parse error in the redshift dialect of sqlfluff.

### Expected Behaviour

I expected this SQL to be parsed without error. It is valid Redshift SQL (indeed, this sql statement is copied from [the AWS documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html#r_CREATE_TABLE_NEW-create-a-table-with-default-identity-column).

### Observed Behaviour

A parse error is thrown when this SQL is linted:

```
$ sqlfluff lint
== [/Users/matt/sandbox/sqlfluff-redshift-mwe/create_table.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE t1(\n hist_id BIGINT IDENTI...'
L:   4 | P:  27 | L005 | Commas should not have whitespace directly before them.
All Finished!
```

Here's the output of running `sqlfluff lint -vvvvv`: [error.log](https://github.com/sqlfluff/sqlfluff/files/8846098/error.log)

### How to reproduce

Minimal example is [in this git repository](https://github.com/menzenski/sqlfluff-redshift-mwe):

`create_table.sql` file:
```sql
CREATE TABLE t1(
  hist_id BIGINT IDENTITY NOT NULL, /* Cannot be overridden */
  base_id BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL, /* Can be overridden */
  business_key varchar(10) ,
  some_field varchar(10)
);
```

`.sqlfluff` file:
```toml
[sqlfluff]
dialect = redshift
```

If the `IDENTITY` / `GENERATED BY DEFAULT AS IDENTITY` are removed, this SQL is parsed without error:

`create_table.sql` file: 
```sql
CREATE TABLE t1(
  hist_id BIGINT NOT NULL, /* Cannot be overridden */
  base_id BIGINT NOT NULL, /* Can be overridden */
  business_key varchar(10) ,
  some_field varchar(10)
);
```

### Dialect

Redshift

### Version

```shell
$ sqlfluff --version
sqlfluff, version 0.13.2
```

```shell
$ python -V
Python 3.9.11
```

### Configuration

`.sqlfluff` file:

```toml
[sqlfluff]
dialect = redshift
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by menzenski** [comment]
Regarding the `Yes I am willing to submit a PR!` - I am willing but I'm new to SQLFluff so would probably need some guidance.

**Comment by tunetheweb** [comment]
This wiki should help with contributing your first change: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

Looks like in `ColumnAttributeSegment` the `IDENTITY` attribute IS supported but only with brackets. Might need to an an `optional=True` attribute to that on line 517 of https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_redshift.py.

**Comment by menzenski** [comment]
@mdahlman I will be glad to open a PR, I should be able to do that today. Thank you!

**Comment by mdahlman** [comment]
Excellent!
If have any troubles finding time for it, you can let me know. Otherwise, I'll leave it to you!
